app.ts

```diff
@@ -0,0 +1,175 @@
+//#region AppLogic
+//#region API methods
+async function getRecordCount() : Promise<number> {
+    const response = await fetch('http://localhost:2050/recordCount');
+    return await response.json();
+}
+async function getColumnNames() : Promise<string[]>{
+    const response = await fetch('http://localhost:2050/columns');
+    return await response.json();
+}
+async function getRecords(fromID: number, toID: number): Promise<string[][]> {
+    const response = await fetch(`http://localhost:2050/records?from=${(fromID).toString()}&to=${(toID).toString()}`);
+    return await response.json();
+}
+//#endregion
+
+//#region Data Loading methods
+
+async function placeRecords(fromID: number, toID: number): Promise<number[]> {
+    const records = await getRecords(fromID, toID)
+    let appendable = '';
+    for (const record of records) {
+            appendable += `<tr id="table-row-${record[0]}">`;
+        for (const column of record) {
+            appendable += `<td align="center">${column.toString()}</td>`;     
+        }
+        appendable += '</tr>';
+    }
+    $("#WrapperTable-ContentBody").empty();
+    $("#WrapperTable-ContentBody").append(appendable);
+    return await [fromID as number, toID as number];
+}
+
+// Credit: https://gist.github.com/ca0v/73a31f57b397606c9813472f7493a940
+
+async function placeRecordsFromCursor(cursor: number[]): Promise<number[]> {
+    cursor = cursor.sort((a,b) => {return a-b});
+    return await placeRecords(cursor[0], cursor[1]);
+}
+//#endregion
+
+//#region Handlers
+async function getPageContent(fromID: number, toID: number): Promise<number[]> {
+    $("#WrapperTable-HeaderRow").empty();
+    getColumnNames()
+        .then((columns: string[]) => {
+            for (const column of columns) {
+                let writable: string = `<th align="center">${column}</th>`;
+                $("#WrapperTable-HeaderRow").append(writable);
+            }
+        });
+    return await placeRecords(fromID, toID);
+}
+
+function toNumber(input: string | number, output: any = 1) : number {
+    switch (typeof input) {
+        case ('string'):
+            if (output == 1) {
+                return parseInt(input as string);
+            } else {
+                return parseFloat(input as string);
+            }
+        case ("number"):
+            return input as number;
+        default:
+            return 0;
+    }
+}
+
+function calculateToId(fromId: number): number {
+    let recordDisplayOffset = 0;
+    if (window.innerHeight <= 646) {
+        recordDisplayOffset = 0
+    } else if (window.innerHeight <= 969) {
+        recordDisplayOffset = 1;
+    } else if (window.innerHeight <= 1938) {
+        recordDisplayOffset = 3
+    } else {
+        recordDisplayOffset = 15
+    }
+
+    return recordDisplayOffset + (fromId + Math.floor((window.innerHeight - ($("#form-content").innerHeight() as number)) / 37));
```

Comment:
There is a little too much logic in this one line for me, being succinct is not good idea when it makes readability difficult. You're going 5 brackets deep in one line here. Rather separate things in to one or two more variables with variable names that describes the logic. A few good variable names often make commenting unnecessary.